### PR TITLE
Page Weight 2022 queries

### PR DESCRIPTION
**Pull request by drohe**
Progress on #2899 

## Assets Section
- [x] page weight distribution
- [x] media page weight over time

### Requests
- [x] distribution of requests
- [x] distribution of requests by content type

### Requests
- [x] distribution of image sizes by format

### Images
- [x] image size distribution

### Videos
- [x] distribution of video sizes by format

## Adoption rate of byte-saving technologies
- [x] % of pages using facades using Lighthouse third-party-facades audit score is not null
- [x] % of pages using compression using Lighthouse uses-text-compression audit score is not null
- [x] % of pages using minification using Lighthouse unminified-javascript & unminified-css score is not null

### CODE REVIEW

**Comment by rviscomi** [comment]
Thanks for working on this @drohe! I've extracted the checklist from the readme and added to the PR comment for easier tracking.

File: sql/2022/page-weight/facades-usage.sql [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    _TABLE_SUFFIX as client,
+    COUNTIF(
+        REGEXP_CONTAINS(report,r'third-party-facades\/\).","score":null')
+    ) AS null_count,
+    COUNTIF(
+        REGEXP_CONTAINS(report,r'third-party-facades\/\).","score":1')
+    ) AS pass_count,
+    COUNTIF(
+        REGEXP_CONTAINS(report,r'third-party-facades\/\).","score":0')
+    ) AS fail_count
```
Comment (rviscomi) [inline sql/2022/page-weight/facades-usage.sql:11]: These `REGEXP_CONTAINS` checks actually work fine, but we should use the semantic JSON functions built into BigQuery. For example:

```suggestion
    COUNTIF(JSON_VALUE(report, '$.audits.third-party-facades.score') IS NULL) AS null_count,
    COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.third-party-facades.score') AS FLOAT64) >= 0.9) AS pass_count,
    COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.third-party-facades.score') AS FLOAT64) < 0.9) AS fail_count
```

This also lets you use 0.9 as the pass/fail threshold, which is more consistent with the way Lighthouse scores audits. In this particular case everything is 0 or null, but it generalizes to all audits.

**Review by rviscomi — DISMISSED** [review]
General comment about parsing Lighthouse results, applies to all LH queries

**Comment by rviscomi** [comment]
@drohe could you take a look at the linter warnings? Hoping to get all analysis PRs merged this week.

**Review by drohe — COMMENTED** [review]
I think I addressed the linting issues and the request to update to use the JSON

**Review by drohe — COMMENTED** [review]
linting updates part 2

**Comment by drohe** [comment]
@rviscomi I'm at a loss on the end of line linting errors from the editorconfig-checker. I've got it set to CRLF in VS Code. Should it be LF?

File: sql/2022/page-weight/bytes_per_type.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  APPROX_QUANTILES(bytesFont / 1024, 1000)[OFFSET(percentile * 10)] AS font_kbytes
+FROM
+  `httparchive.summary_pages.2022_06_01_*`,
+  UNNEST([10, 25, 50, 75, 90, 100]) AS percentile
+GROUP BY
+  percentile,
+  client
+ORDER BY
+  client,
+  percentile
```
Comment (rviscomi) [inline sql/2022/page-weight/bytes_per_type.sql:20]:
```suggestion
  percentile

```

File: sql/2022/page-weight/compression-usage.sql [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) < 0.9) AS fail_count
+FROM
+  `httparchive.lighthouse.2022_06_01_*`
+GROUP BY
+  client
+ORDER BY
+  client,
+  null_count,
+  pass_count,
+  fail_count
```
Comment (rviscomi) [inline sql/2022/page-weight/compression-usage.sql:14]:
```suggestion
  fail_count

```

File: sql/2022/page-weight/facades-usage.sql [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.third-party-facades.score') AS FLOAT64) < 0.9) AS fail_count
+FROM
+  `httparchive.lighthouse.2022_06_01_*`
+GROUP BY
+  client
+ORDER BY
+  client,
+  null_count,
+  pass_count,
+  fail_count
```
Comment (rviscomi) [inline sql/2022/page-weight/facades-usage.sql:14]:
```suggestion
  fail_count

```

File: sql/2022/page-weight/minified_css_usage.sql [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.unminified-css.score') AS FLOAT64) < 0.9) AS fail_count
+FROM
+  `httparchive.lighthouse.2022_06_01_*`
+GROUP BY
+  client
+ORDER BY
+  client,
+  null_count,
+  pass_count,
+  fail_count
```
Comment (rviscomi) [inline sql/2022/page-weight/minified_css_usage.sql:14]:
```suggestion
  fail_count

```

File: sql/2022/page-weight/minified_js_usage.sql [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.unminified-javascript.score') AS FLOAT64) < 0.9) AS fail_count
+FROM
+  `httparchive.lighthouse.2022_06_01_*`
+GROUP BY
+  client
+ORDER BY
+  client,
+  null_count,
+  pass_count,
+  fail_count
```
Comment (rviscomi) [inline sql/2022/page-weight/minified_js_usage.sql:14]:
```suggestion
  fail_count

```

File: sql/2022/page-weight/page_weight_trend.sql [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  `httparchive.summary_pages.*`
+WHERE
+  bytesTotal > 0 AND
+  SUBSTR(_TABLE_SUFFIX, 9, 2) = '01' -- ignore mid-month figures as not always available and throws off chart
+GROUP BY
+  date,
+  client
+ORDER BY
+  date DESC,
+  client
```
Comment (rviscomi) [inline sql/2022/page-weight/page_weight_trend.sql:20]:
```suggestion
  client

```

File: sql/2022/page-weight/request_type_distribution.sql [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  APPROX_QUANTILES(reqFont, 1000)[OFFSET(percentile * 10)] AS font_req
+FROM
+  `httparchive.summary_pages.2022_06_01_*`,
+  UNNEST([10, 25, 50, 75, 90, 100]) AS percentile
+GROUP BY
+  percentile,
+  client
+ORDER BY
+  client,
+  percentile
```
Comment (rviscomi) [inline sql/2022/page-weight/request_type_distribution.sql:20]:
```suggestion
  percentile

```

File: sql/2022/page-weight/response_format_distribution.sql [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+  `httparchive.summary_requests.2022_06_01_*`,
+  UNNEST([10, 25, 50, 75, 90, 100]) AS percentile
+GROUP BY
+  client,
+  percentile,
+  format
+ORDER BY
+  format,
+  client,
+  percentile
```
Comment (rviscomi) [inline sql/2022/page-weight/response_format_distribution.sql:18]:
```suggestion
  percentile

```

File: sql/2022/page-weight/response_type_distribution.sql [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+  `httparchive.summary_requests.2022_06_01_*`,
+  UNNEST([10, 25, 50, 75, 90, 100]) AS percentile
+GROUP BY
+  client,
+  percentile,
+  type
+ORDER BY
+  client,
+  type,
+  percentile
```
Comment (rviscomi) [inline sql/2022/page-weight/response_type_distribution.sql:16]:
```suggestion
  percentile

```

**Review by rviscomi — COMMENTED** [review]
Weird I've never seen that warning before. Let's see if deleting/adding the newline from code review helps.

**Comment by rviscomi** [comment]
Ok that did the trick! In the GitHub editor (press `.`) I changed CRLF to LF.

File: sql/2022/page-weight/compression-usage.sql [hunk 11]
```diff
@@ -0,0 +1,14 @@
+SELECT
+  _TABLE_SUFFIX AS client,
+  COUNTIF(JSON_VALUE(report, '$.audits.text-compression.score') IS NULL) AS null_count,
+  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) >= 0.9) AS pass_count,
+  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) < 0.9) AS fail_count
```
Comment (rviscomi) [inline sql/2022/page-weight/compression-usage.sql:5]: For each of these "count" queries, we should also calculate the percent of pages. Something like this:
```suggestion
  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) >= 0.9) AS pass_count,
  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) < 0.9) AS fail_count,
  COUNT(0) AS total,
  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) >= 0.9) / COUNT(0) AS pct_pass,
  COUNTIF(SAFE_CAST(JSON_VALUE(report, '$.audits.text-compression.score') AS FLOAT64) < 0.9) / COUNT(0) AS pct_fail
```

Can you also update the rest of the queries to calculate percents?

File: sql/2022/page-weight/page_weight_trend.sql [hunk 12]
```diff
@@ -0,0 +1,20 @@
+#standardSQL
+SELECT
+  REPLACE(SUBSTR(_TABLE_SUFFIX, 0, 10), '_', '') AS date,
```
Comment (rviscomi) [inline sql/2022/page-weight/page_weight_trend.sql:3]: Not sure why but YYYY-MM-DD format seems to work best for custom date formatting in sheets, ie to display as "Aug 2022".
```suggestion
  REPLACE(SUBSTR(_TABLE_SUFFIX, 0, 10), '_', '-') AS date,
```

**Review by rviscomi — CHANGES_REQUESTED** [review]

File: sql/2022/page-weight/compression-usage.sql (hunk 11 above)
Comment (rviscomi) [reply sql/2022/page-weight/compression-usage.sql:5]: @drohe let me know if I can help at all with this. Want to make sure we give @fellowhuman1101 enough time to write the draft.

**Comment by drohe** [comment]
added the percentages

**Review by rviscomi — APPROVED** [review]

**Pull request merged**